lib/screen/detail/preview/canvas_device_bloc.dart

```diff
@@ -113,38 +115,30 @@ class CanvasDeviceCastDailyWorkEvent extends CanvasDeviceEvent {
 class CanvasDeviceState {
   final List<DeviceState> devices;
   final Map<String, CheckDeviceStatusReply> canvasDeviceStatus;
-  final Map<String, CanvasDevice> lastSelectedActiveDeviceMap;
+
+  final _selectedCanvasDeviceStore = injector<SelectedCanvasDeviceStore>();
 
   // final String sceneId;
   final RPCError? rpcError;
 
   CanvasDeviceState({
     required this.devices,
     Map<String, CheckDeviceStatusReply>? canvasDeviceStatus,
-    Map<String, CanvasDevice>? lastSelectedActiveDeviceMap,
     this.rpcError,
-  })  : canvasDeviceStatus = canvasDeviceStatus ?? {},
-        lastSelectedActiveDeviceMap = lastSelectedActiveDeviceMap ?? {};
+  }) : canvasDeviceStatus = canvasDeviceStatus ?? {};
 
   CanvasDeviceState copyWith(
           {List<DeviceState>? devices,
           Map<String, CheckDeviceStatusReply>? controllingDeviceStatus,
-          Map<String, CanvasDevice>? lastActiveDevice,
           RPCError? rpcError}) =>
       CanvasDeviceState(
           devices: devices ?? this.devices,
           canvasDeviceStatus: controllingDeviceStatus ?? canvasDeviceStatus,
-          lastSelectedActiveDeviceMap:
-              lastActiveDevice ?? lastSelectedActiveDeviceMap,
           rpcError: rpcError ?? this.rpcError);
 
-  CanvasDeviceState updateOnCast(
-      {required CanvasDevice device, required String displayKey}) {
-    lastSelectedActiveDeviceMap.removeWhere((key, value) => value == device);
-    lastSelectedActiveDeviceMap[displayKey] = device;
-    return copyWith(
-      lastActiveDevice: lastSelectedActiveDeviceMap,
-    );
+  Future<void> updateOnCast(
+      {required CanvasDevice device, required String displayKey}) async {
+    await _selectedCanvasDeviceStore.save(device, displayKey);
```

Comment:
Did you forget `lastSelectedActiveDeviceMap.removeWhere((key, value) => value == device);`?